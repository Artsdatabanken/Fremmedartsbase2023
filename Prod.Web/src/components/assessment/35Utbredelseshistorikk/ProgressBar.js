import React from 'react'

const ProgressBar = ({
        prefixCls, strokeWidth, trailWidth, strokeColor,
        trailColor, strokeLinecap, percent, style, className,
    }) => {

    const pathStyle = {
        strokeDasharray: '100px, 100px',
        strokeDashoffset: `${(100 - percent)}px`,
        transition: 'stroke-dashoffset 0.6s ease 0s, stroke 0.6s linear',
    }

    const center = strokeWidth / 2
    const right = 100 - (strokeWidth / 2)
    const pathString = `M ${center},${center} L ${right},${center}`
    const viewBoxString = `0 0 100 ${strokeWidth}`

    return (
        <svg
            className={`${prefixCls}-line ${className}`}
            viewBox={viewBoxString}
            preserveAspectRatio="none"
            style={style}>
            <path
                className={`${prefixCls}-line-trail`}
                d={pathString}
                strokeLinecap={strokeLinecap}
                stroke={trailColor}
                strokeWidth={trailWidth || strokeWidth}
                fillOpacity="0"/>
            <path
                className={`${prefixCls}-line-path`}
                d={pathString}
                strokeLinecap={strokeLinecap}
                stroke={strokeColor}
                strokeWidth={strokeWidth}
                fillOpacity="0"
                style={pathStyle}/>
        </svg>
    )
}

ProgressBar.defaultProps = {
    prefixCls: 'rc-progress',
    strokeWidth: 1,
    strokeColor: '#039be5',
    trailWidth: 1,
    trailColor: '#D9D9D9',
    strokeLinecap: 'round',
    className: ''
}

ProgressBar.propTypes = {
    prefixCls: React.PropTypes.string,
    strokeWidth: React.PropTypes.oneOfType([React.PropTypes.number, React.PropTypes.string]),
    strokeColor: React.PropTypes.string,
    trailWidth: React.PropTypes.oneOfType([React.PropTypes.number, React.PropTypes.string]),
    trailColor: React.PropTypes.string,
    strokeLinecap: React.PropTypes.oneOf(['round', 'square']),
    style: React.PropTypes.object,
    className: React.PropTypes.string,
}

export default ProgressBar
