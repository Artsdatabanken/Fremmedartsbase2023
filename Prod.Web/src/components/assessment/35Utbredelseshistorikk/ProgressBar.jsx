import React from 'react'
import PropTypes from 'prop-types'


const ProgressBar = ({
    prefixCls, strokeWidth, trailWidth, strokeColor,
    trailColor, strokeLinecap, percent, style, className,
}) => {

    const pathStyle = {
        strokeDasharray: '100px, 100px',
        strokeDashoffset: `${(100 - percent)}px`,
        transition: 'stroke-dashoffset 0.6s ease 0s, stroke 0.6s linear',
    }

    const center = strokeWidth / 2
    const right = 100 - (strokeWidth / 2)
    const pathString = `M ${center},${center} L ${right},${center}`
    const viewBoxString = `0 0 100 ${strokeWidth}`

    return (
        <svg
            className={`${prefixCls}-line ${className}`}
            viewBox={viewBoxString}
            preserveAspectRatio="none"
            style={style}>
            <path
                className={`${prefixCls}-line-trail`}
                d={pathString}
                strokeLinecap={strokeLinecap}
                stroke={trailColor}
                strokeWidth={trailWidth || strokeWidth}
                fillOpacity="0" />
            <path
                className={`${prefixCls}-line-path`}
                d={pathString}
                strokeLinecap={strokeLinecap}
                stroke={strokeColor}
                strokeWidth={strokeWidth}
                fillOpacity="0"
                style={pathStyle} />
        </svg>
    )
}

ProgressBar.defaultProps = {
    prefixCls: 'rc-progress',
    strokeWidth: 1,
    strokeColor: '#039be5',
    trailWidth: 1,
    trailColor: '#D9D9D9',
    strokeLinecap: 'round',
    className: ''
}

ProgressBar.propTypes = {
    prefixCls: PropTypes.string,
    strokeWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    strokeColor: PropTypes.string,
    trailWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    trailColor: PropTypes.string,
    strokeLinecap: PropTypes.oneOf(['round', 'square']),
    style: PropTypes.object,
    className: PropTypes.string,
}

export default ProgressBar
