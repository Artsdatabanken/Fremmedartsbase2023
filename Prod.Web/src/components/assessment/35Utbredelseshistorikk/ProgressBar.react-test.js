import React from 'react'
import renderer from 'react-test-renderer'
import ProgressBar from './ProgressBar'

function assert(targetComponment) {
    const component = renderer.create(targetComponment)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
}

test('Progress bar 0%', () => {
    assert(<ProgressBar percent={0} />)
})

test('Progress bar 100%', () => {
    assert(<ProgressBar percent={100} />)
})
