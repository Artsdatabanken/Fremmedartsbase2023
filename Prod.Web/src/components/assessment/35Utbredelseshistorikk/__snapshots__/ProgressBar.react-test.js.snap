exports[`test Progress bar 0% 1`] = `
<svg
  className="rc-progress-line "
  preserveAspectRatio="none"
  style={undefined}
  viewBox="0 0 100 1">
  <path
    className="rc-progress-line-trail"
    d="M 0.5,0.5 L 99.5,0.5"
    fillOpacity="0"
    stroke="#D9D9D9"
    strokeLinecap="round"
    strokeWidth={1} />
  <path
    className="rc-progress-line-path"
    d="M 0.5,0.5 L 99.5,0.5"
    fillOpacity="0"
    stroke="#039be5"
    strokeLinecap="round"
    strokeWidth={1}
    style={
      Object {
        "strokeDasharray": "100px, 100px",
        "strokeDashoffset": "100px",
        "transition": "stroke-dashoffset 0.6s ease 0s, stroke 0.6s linear",
      }
    } />
</svg>
`;

exports[`test Progress bar 100% 1`] = `
<svg
  className="rc-progress-line "
  preserveAspectRatio="none"
  style={undefined}
  viewBox="0 0 100 1">
  <path
    className="rc-progress-line-trail"
    d="M 0.5,0.5 L 99.5,0.5"
    fillOpacity="0"
    stroke="#D9D9D9"
    strokeLinecap="round"
    strokeWidth={1} />
  <path
    className="rc-progress-line-path"
    d="M 0.5,0.5 L 99.5,0.5"
    fillOpacity="0"
    stroke="#039be5"
    strokeLinecap="round"
    strokeWidth={1}
    style={
      Object {
        "strokeDasharray": "100px, 100px",
        "strokeDashoffset": "0px",
        "transition": "stroke-dashoffset 0.6s ease 0s, stroke 0.6s linear",
      }
    } />
</svg>
`;
